src/main/cpp/include/utils/alleles_combiner_template_definition.h

```diff
@@ -316,3 +427,27 @@ void AllelesCombiner::get_merged_VCF_spec_alleles_vec(std::string& buffer, std::
     alleles_vec.emplace_back(&(buffer[begin_idx]), buffer.size()-begin_idx);
   }
 }
+
+template<typename ValidRowTrackerTy>
+int AllelesCombiner<ValidRowTrackerTy>::get_merged_allele_idx(const size_t row_query_idx, const int allele_idx) const {
+  switch((is_REF_block(row_query_idx) << 1u)
+      | (contains_NON_REF_allele(row_query_idx))) {
+    case 0u:
+      return get_merged_allele_idx<true, false, false>(row_query_idx, allele_idx);
+    case 1u:
+      return get_merged_allele_idx<true, false, true>(row_query_idx, allele_idx);
+    case 2u:
+      throw AllelesCombinerException(
```

Comment:
If possible use `logger.fatal` and `logger.format` for conciseness and readability.